File: MISC/convert.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s
        
        L = [''] * numRows
        index, step = 0, 1

        for ch in s:
            L[index] += ch
            if index == 0:
                step = 1
            elif index == numRows -1:
                step = -1
            index += step
        return ''.join(L)
            

File: MISC/Convert.cs
public class Solution {
    public string Convert(string s, int numRows) {
        // return s in case we have only one row
        if (numRows == 1) return s;

        StringBuilder ret = new StringBuilder();
        int n = s.Length;
        int cycleLen = 2 * numRows - 2;

        for (int i = 0; i < numRows; i++) {
            for (int j = 0; j + i < n; j += cycleLen) {
                ret.Append(s[j + i]);
                if (i != 0 && i != numRows - 1 && j + cycleLen - i < n)
                    ret.Append(s[j + cycleLen - i]);
            }
        }
        return ret.ToString();
    }
}